server.go

```diff
@@ -163,31 +164,41 @@ func newServer(listenAddrs []string, notifier chainntnfs.ChainNotifier,
 		return nil, fmt.Errorf("default listener must be TCP")
 	}
 
-	// TODO(roasbeef): remove once we actually sign the funding_locked
-	// stuffs
-	fakeSigHex, err := hex.DecodeString("30450221008ce2bc69281ce27da07e66" +
-		"83571319d18e949ddfa2965fb6caa1bf0314f882d70220299105481d63e0f" +
-		"4bc2a88121167221b6700d72a0ead154c03be696a292d24ae")
-	if err != nil {
-		return nil, err
-	}
+	chanGraph := chanDB.ChannelGraph()
 
-	fakeSig, err := btcec.ParseSignature(fakeSigHex, btcec.S256())
+	// In order to have ability to announce the self node we need to
+	// sign the node announce message, and include the signature in the
+	// node channeldb object.
+	// TODO(andrew.shvv) possible sign message on every announcement in
+	// order to avoid such ugliness?
+	alias, err := lnwire.NewAlias(hex.EncodeToString(serializedPubKey[:10]))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("can't create alias: %v", err)
 	}
-
-	chanGraph := chanDB.ChannelGraph()
 	self := &channeldb.LightningNode{
-		AuthSig:    fakeSig,
 		LastUpdate: time.Now(),
 		Address:    selfAddr,
 		PubKey:     privKey.PubKey(),
 		// TODO(roasbeef): make alias configurable
-		Alias: hex.EncodeToString(serializedPubKey[:10]),
+		Alias: alias.String(),
 	}
+
+	// Initialize graph with authenticated lightning node, signature is
+	// needed in order to be able to announce node to other network.
+	messageSigner := lnwallet.NewMessageSigner(s.identityPriv)
+	if self.AuthSig, err = discovery.SignAnnouncement(messageSigner,
```

Comment:
Why's this going through the discovery service unnecessarily? Instead (taking into account my prior comments), the signing should be done in-place here.